R/model.R

```diff
@@ -848,6 +852,7 @@ sample_method <- function(data = NULL,
   checkmate::assert_integerish(chains, lower = 1, len = 1)
   checkmate::assert_integerish(parallel_chains, lower = 1, null.ok = TRUE)
   checkmate::assert_integerish(threads_per_chain, lower = 1, len = 1, null.ok = TRUE)
+  checkmate::assert_integerish(chain_id, lower = 1, len = chains, unique = TRUE, null.ok = FALSE)
```

Comment:
Can we just fall back on the check on `proc_ids` when validating the CmdStanArgs object? https://github.com/stan-dev/cmdstanr/blob/bc3c8a7824fcd13184fe447e73dfbbcc295eec83/R/args.R#L453

Or do we need to check it immediately here to avoid an issue before it's checked later?

Reply:
This does some additional checks. And I think the error mentioning `'chain_ids'` as opposed to `'sefl$procs_ids'` is more informative to the user.

Reply:
Ok yeah that makes sense.